### PR TITLE
Update gleam-for-elixir-users.md

### PR DESCRIPTION
**Pull request by MiniAppleTheApple**
# Todo

- [ ] Flow Control 
  - [ ] Case
  - [ ] Try
- [ ] Type aliases
- [ ] Custom types
  - [ ] Records
  - [ ] Unions
  - [ ] Opaque custom types
- [ ] Modules
  - [ ] Imports
  - [ ] Nested modules
  - [ ] First class modules

### CODE REVIEW

File: cheatsheets/gleam-for-elixir-users.md [hunk 1]
```diff
@@ -659,3 +659,32 @@ pub fn main() {
...
+## Patterns
+
+Same as Elixir, Gleam has pattern matching, which is used for matching complex structured data. In Gleam we use `as` to name the variable, same as using `=` in Elixir.
+
+#### Gleam
+
+```gleam
+let list = [1, 2, 3]
+
+let assert [first as 1, second .. rest] = list
```
Comment (lpil) [inline cheatsheets/gleam-for-elixir-users.md:671]:
```suggestion
let assert [1 as first, second, ..rest] = list
```

File: cheatsheets/gleam-for-elixir-users.md [hunk 2]
```diff
@@ -659,3 +659,32 @@ pub fn main() {
...
+
+let assert Foo(foo: 10 as foo, bar: 20 as bar) = Foo(foo: 10, bar: 20)
```
Comment (lpil) [inline cheatsheets/gleam-for-elixir-users.md:673]: No "foo" please, it's a rather depressing WW2 reference.

**Comment by lpil** [comment]
Would be good to explain that Gleam has exhaustiveness checking here

File: cheatsheets/gleam-for-elixir-users.md (hunk 2 above)
Comment (inoas) [reply cheatsheets/gleam-for-elixir-users.md:673]: I did not know till now, no more foo/foobar anymore for me then.

Comment (MiniAppleTheApple) [reply cheatsheets/gleam-for-elixir-users.md:673]: Sure, thanks for suggesting

**Comment by MiniAppleTheApple** [comment]
The idea of using let assert is to let the readers know the equivalent feature in Gleam. However, I will address the exhaustiveness checking you mentioned and explain to the readers a more "Gleam" way of thinking instead of the "Elixir" way.

**Pull request merged**